sigmadsp/hardware/dsp.py

```diff
@@ -101,6 +109,13 @@ def parse_config(self):
         except (KeyError, TypeError):
             logger.info("No DSP pin definitions were found in the configuration file.")
 
+        self.type = self.config["dsp"]["type"]
+        self.protocol = self.config["dsp"].get("protocol", "spi")
```

Comment:
I would suggest to add type hints for these two fields in the `__init__` function, just for readability.

Regarding the default (`spi`), I prefer to throw an error and let the user correct their config file. The protocol should not be an optional parameter.

Reply:
I went for the thing that would break the fewest installations :)

going to add errors and make configuration options more uniform (`bus_number` and `device_address` instead of `i2c_bus`, `i2c_address`, `spi_bus`, `spi_address`)

Reply:
You are right, I didn't even think about existing installations. In that case, this will be a breaking release, so it is more obvious.